pkg/components/velero/component.go

```diff
@@ -201,14 +201,14 @@ func (c *component) getProvider() (provider, error) {
 		providers = append(providers, c.Restic)
 	}
 
-	if len(providers) > 1 {
-		return nil, fmt.Errorf("more than one provider configured")
-	}
-
 	if len(providers) == 0 {
 		return nil, fmt.Errorf("no providers configured")
 	}
 
+	if len(providers) > 1 {
+		return nil, fmt.Errorf("more than one provider configured")
+	}
+
```

Comment:
We can get rid of two if blocks and simply check if the providers length is not 1 then error out something like `expected 1 provider got %d provider(s)`.